server/container/core/src/main/java/org/apache/james/core/MailImpl.java

```diff
@@ -411,7 +413,7 @@ public void setMessage(MimeMessage message) {
      */
     @Override
     public void setRecipients(Collection<MailAddress> recipients) {
-        this.recipients = new ArrayList<MailAddress>(recipients);
+        this.recipients = new HashSet<MailAddress>(recipients);
```

Comment:
No idea... 

Maybe a TreeSet should be more appropriate then, let's be conservative about the order...